src/app/integrations/list-page/list-page.component.html

```diff
@@ -4,15 +4,72 @@
                        [viewTemplate]="viewTemplate"
                        [filterFields]="filterFields">
   <ng-template #viewTemplate>
-    <button type="button"
-            class="btn btn-primary toolbar-pf-action-right"
-            [routerLink]="['/integrations/create']">
-            Create Integration
-    </button>
-  </ng-template>
+    <div class="toolbar-pf-action-right">
+        <button type="button"
+                class="btn btn-default"
+                (click)="showImportDialog()">
+                Import
+        </button>
+        <button type="button"
+                class="btn btn-primary"
+                [routerLink]="['/integrations/create']">
+                Create Integration
+        </button>
+    </div>
+
+</ng-template>
 </syndesis-list-toolbar>
 
 <syndesis-loading [loading]="loading | async">
   <syndesis-integrations-list [complete]="true"
                               [integrations]="filteredIntegrations | async"></syndesis-integrations-list>
 </syndesis-loading>
+
+<syndesis-modal id="importIntegration" title="Integration Import" [body]="importIntegrationModal">
+
+  <ng-template #importIntegrationModal let-modal="modal">
+
+    Select the integration file to import:
+    <input #fileSelect type="file" ng2FileSelect [uploader]="uploader" (onFileSelected)="fileSelect.value=''" />
+
+    <!-- for more info/examples of how to use the ng2 file uploader, see https://valor-software.com/ng2-file-upload/ -->
+    <div ng2FileDrop
+      [uploader]="uploader"
+      class="well my-drop-zone">
+      Or drag and drop your integration files here
```

Comment:
Awesome :-)

Reply:
@gashcrumb   Wonder whether we could introduce feature toggles in the UI, too ? So that we could switch off the feature based on some config variable. I have still the hope that we can avoid branching when we e.g. have to adapt TP2 based installations.

Reply:
That would be cool, don't see why we couldn't do this.

Reply:
@rhuss no reason we can't, I'd probably suggest driving that via the `config.json`

Reply:
Perhaps even derive the feature toggles based on what is configured in the rest backend since type typically work hand in hand.